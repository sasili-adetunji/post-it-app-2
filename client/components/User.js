import React from 'react';
import { ListItem } from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import API from '../Api';

class User extends React.Component {
  /**
   * Creates an instance of User.
   * @param {any} props
   * @memberof User
   */
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div>
        <ListItem
        href={`/#/dashboard/users/${this.props.user.userId}`}
      > {this.props.user.username} </ListItem>
      </div>
    );
  }
}

export default User;
