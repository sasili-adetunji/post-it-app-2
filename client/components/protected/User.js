import React from 'react';

/**
 *
 * @class Message
 * @extends {React.Component}
 */
class Message extends React.Component {
  constructor(props) {
    super(props);
  }

  /**
   *
   * renders the message components
   * @returns {void}
   * @memberof Message
   */

  render() {
    return (
      <div className="side-menu-container">
        {/* <ul className="nav navbar-nav"> */}
          <li><span />
            {this.props.user.userName} </li>
        {/* </ul> */}
      </div>
    );
  }
}
export default Message;
