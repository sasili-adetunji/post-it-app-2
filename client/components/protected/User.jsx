import React from 'react';

class Message extends React.Component {

  /**
 * Displays the List of Users in a group
 * 
 * @param props
 * 
 * @class User
 * 
 * @extends {Component}
 */

  render() {
    return (
      <div className="side-menu-container list-group-item">
        <li><span />
          <b> {this.props.user.userName} </b> </li>
      </div>
    );
  }
}
export default Message;
