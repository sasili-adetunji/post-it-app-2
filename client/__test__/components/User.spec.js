import React from 'react';
import { shallow, render } from 'enzyme';
import User from '../../components/protected/User.jsx';

describe('User components', () => {
  const user = {
    userName: 'sasil',
    email: 'sasil@yahoo.com',
    id: 'UTsUlauteS5nb34huOP6SpvcuocX9vA1',
  };

  const component = shallow(<User user={user} />);

  it('should match snapshot test', () => {
    expect(component).toMatchSnapshot();
  });
  it('should render', () => {
    expect(component).toBeDefined();
  });
  it('Should contain one div', () => {
    expect(component.find('div').length).toEqual(1);
  });
  it('should recieve props', () => {
    expect(Object.keys(component.props()).length).toBeGreaterThan(0);
  });
  it('should render without throwing an error', () => {
    expect(component.contains(user.userName)).toBe(true);
  });
});
