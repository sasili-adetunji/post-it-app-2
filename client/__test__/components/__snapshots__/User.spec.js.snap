// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User components should match snapshot test 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="side-menu-container list-group-item"
  >
    <li>
      <span />
      <b>
         
        sasil
         
      </b>
       
    </li>
  </div>,
  "nodes": Array [
    <div
      className="side-menu-container list-group-item"
    >
      <li>
        <span />
        <b>
           
          sasil
           
        </b>
         
      </li>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Message
      user={
        Object {
          "email": "sasil@yahoo.com",
          "id": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
          "userName": "sasil",
        }
      }
    />,
    "_instance": Message {
      "context": Object {},
      "props": Object {
        "user": Object {
          "email": "sasil@yahoo.com",
          "id": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
          "userName": "sasil",
        },
      },
      "refs": Object {},
      "state": Object {},
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <div
      className="side-menu-container list-group-item"
    >
      <li>
        <span />
        <b>
           
          sasil
           
        </b>
         
      </li>
    </div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Message
    user={
      Object {
        "email": "sasil@yahoo.com",
        "id": "UTsUlauteS5nb34huOP6SpvcuocX9vA1",
        "userName": "sasil",
      }
    }
  />,
}
`;
